Tools/Scripts/libraries/webkitapipy/webkitapipy/tbd.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (C) 2025 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import annotations
+from typing import Iterable
+from pathlib import Path
+
+
+class _TBD:
+    @property
+    def install_name(self):
+        return getattr(self, 'install-name')
+
+    @classmethod
+    def load_from(cls, file: Path) -> Iterable[_TBD]:
+        raise NotImplementedError
+
+
+TBD: type[_TBD]
+
+try:
+    import yaml
+
+    class _PyYAMLTBD(_TBD, yaml.YAMLObject):
+        yaml_tag = '!tapi-tbd'
+
+        @classmethod
+        def load_from(cls, file):
+            with open(file) as fd:
+                for tbd in yaml.unsafe_load_all(fd):
+                    yield tbd
+    TBD = _PyYAMLTBD
+
+except ImportError:
```

Comment:
Do we actually care about this working if PyYAML isn't installed? If we believe the `pyproject.toml` file, PyYAML is a requirement. Otherwise, we should change the dependencies, potentially adding `tbd` as an extra, which requires PyYAML?

Reply:
We care about the being able to use the underlying functionality (being able to load TBDs from an SDK) in cloud-based environments where we can't use the autoinstaller. Point taken on the pyproject.toml spec not accurately reflecting that.

Given this and below feedback, I'm tempted to switch to an alternative implementation for TBD reading. I learned that `llvm-readtapi` (available in Xcode 16+) can convert YAML-based TBDs to a JSON representation, so by switching to it we could replace the PyYAML logic entirely.